pandas/tests/io/parser/test_dtypes.py

```diff
@@ -550,3 +550,19 @@ def test_numeric_dtype(all_parsers, dtype):
 
     result = parser.read_csv(StringIO(data), header=None, dtype=dtype)
     tm.assert_frame_equal(expected, result)
+
+
+@pytest.mark.parametrize("na_string", ["NaN", "nan", "NA", "null", "NULL", ""])
+def test_boolean_dtype(all_parsers, na_string):
+    parser = all_parsers
+    data = f"a,b\nTrue,False\nTrue,{na_string}\n"
```

Comment:
Can you assert that all the true / false strings are testsed?

Reply:
I'll parametrize this over the different strings as well